### ISSUE TITLE
Carthage: dyld: Library not loaded: @rpath/Mixpanel.framework/Mixpanel

### ISSUE THREAD

**Issue by vandutran** [mixpanel/mixpanel-iphone#345]
Got this error when running my app.

dyld: Library not loaded: @rpath/Mixpanel.framework/Mixpanel
Reason: image not found

Xcode 6.4, mixpanel 2.8.3, dragged carthage generated .framework file to project. No compile errors.

**Comment by brennanMKE** [comment]
This happened to my project as well.

**Comment by brennanMKE** [comment]
I resolved this problem with my project. I went to the General tab for the build target and added the Mixpanel framework in the Embedded Binaries section. I did find that it then doubled up the reference to Mixpanel in the frameworks so I removed it from both and added it again to Embedded Binaries and it only had one reference in both sections. Now the app runs without any errors.

@vandutran

**Comment by samgreen** [comment]
@vandutran Is this still an issue for you?

**Comment by vandutran** [comment]
I stopped using the Carthage framework after getting this error, now i'm importing the whole source code (.h, .m)

Haven't retried the carthage framework, but i'll give it another try later.

**Comment by samgreen** [comment]
Bumping this to see if v2.9.0 resolves this @vandutran

**Comment by vandutran** [comment]
Works now. Thanks.

**Comment by vandutran** [comment]
I spoke too fast, got a crash at launch.

dyld: Library not loaded: @rpath/Mixpanel.framework/Mixpanel
Referenced from: /var/mobile/Containers/Bundle/Application/007B2C51-2BEA-466A-AFB9-A332C57192FB/xxx.app/xxx
Reason: image not found

Xcode 7

**Comment by samgreen** [comment]
@vandutran Can you try an experiment and link to the framework weakly?

I'm not having that issue in our internal carthage test project.

**Comment by vandutran** [comment]
I made the framework "Optional" in Linked Frameworks and Librairies and I get an exception when it reaches the Mixpanel.sharedInstance() line.

Adding the framework in "Embedded Binaries" like @brennanMKE suggested works. But I don't know if that's the right way to do things?

It's pretty simple to reproduce, you can just create a blank swift project, carthage mixpanel, and add it to the project.

**Comment by jaredmixpanel** [comment]
I also ran into this issue and was able to fix it by adding a Copy Files Build Phase, setting its destination to Frameworks and adding the Mixpanel framework there (in addition to Link Binary with Libraries), like this: 
![carthage_test](https://cloud.githubusercontent.com/assets/10504508/10698571/f6884f88-79b1-11e5-8d16-d2f40c03ff05.png)

As discussed here: https://github.com/Carthage/Carthage/issues/615#issuecomment-120927579

**Comment by samgreen** [comment]
Are you making use of the copy-frameworks script that Carthage recommends in the README? I'm using it in our test project and have never seen this error:

On your application targets’ “Build Phases” settings tab, click the “+” icon and choose “New Run Script Phase”. Create a Run Script with the following contents:

/usr/local/bin/carthage copy-frameworks
and add the paths to the frameworks you want to use under “Input Files”, e.g.:

$(SRCROOT)/Carthage/Build/iOS/Box.framework
$(SRCROOT)/Carthage/Build/iOS/Result.framework
$(SRCROOT)/Carthage/Build/iOS/ReactiveCocoa.framework

**Comment by samgreen** [comment]
Closing as inactive. Seems like the framework needs to be embedded. If anyone has questions, see @jaredmixpanel answer.

**Comment by obi55** [comment]
@brennanMKE 's answer did the trick!

**Comment by mobileben** [comment]
Is this actually still an issue? I did a manual add. Followed the instructions but ended up with 

> dyld: Library not loaded: @rpath/Mixpanel.framework/Mixpanel

I had to embed the framework. Other pre-built 3rd Party SDKs I use haven't needed this.

**Comment by zihejia** [comment]
Hi @mobileben the latest document does mention you need to embed the framework
`Step 3: Embed the Mixpanel framework
Select your app .xcodeproj file. Under "General", add the Mixpanel framework as an embedded binary. Once added, please make sure Mixpanel.framework shows under both "Linked Frameworks and Libaries" and "Embedded Binaries". `

**Comment by mobileben** [comment]
@zihejia Ahh, sorry my bad. Is this based on the GitHub page?? I didn't see any instructions regarding this. I went through the website and ended up here (it's an old link).

https://mixpanel.com/blog/2014/09/30/integrating-the-mixpanel-ios-sdk-without-cocoapods/

But I got to it by way of the main instructions here:

https://developer.mixpanel.com/docs/ios

Note that I bypassed the main GitHub page for the project since I got the latest from here:

https://github.com/mixpanel/mixpanel-iphone/releases

And downloaded the framework. 

So it may be helpful to update the link for "manually install the Mixpanel library".

Seems I fall into the edge case path of getting the SDK.
